packages/build/src/log/messages/core_commands.js

```diff
@@ -1,17 +1,72 @@
 'use strict'
 
-const { log, logArray, logError } = require('../logger')
+const path = require('path')
+
+const { log, logArray, logErrorSubHeader } = require('../logger')
 const { THEME } = require('../theme')
 
-const logExperimentalEsbuildParameter = function (logs, envVariableName) {
-  logError(
-    logs,
-    `
-You've opted in to an experimental bundling mechanism via the "${envVariableName}" environment variable.
-We recommend against using this functionality in production sites since its behavior may change or it may be removed entirely.
-For the latest updates about function bundling, please visit our Community forum: community.netlify.com.
-`,
-  )
+const logBundleErrorObject = ({ logs, object }) => {
+  const { location = {}, text } = object
+  const { column, file, line, lineText } = location
+  const locationMessage = `${file} ${line}:${column}`
+  const logMessage = `> ${lineText}
+  (in ${locationMessage})
+
+  ${text}
+  
+  `
+
+  log(logs, logMessage, { indent: true })
+}
+
+const logEsbuildBundleResult = ({ functionName, logs, result }) => {
+  const { bundlerErrors = [], bundlerWarnings = [] } = result
+
+  if (bundlerErrors.length !== 0) {
+    logErrorSubHeader(logs, `Bundling function \`${functionName}\` has failed:`)
+
+    bundlerErrors.forEach((bundlerError) => {
+      logBundleErrorObject({ logs, object: bundlerError })
+    })
+  }
+
+  if (bundlerWarnings.length !== 0) {
+    logErrorSubHeader(logs, `Bundling function \`${functionName}\` has produced some warnings:`)
+
+    bundlerWarnings.forEach((bundlerWarning) => {
+      logBundleErrorObject({ logs, object: bundlerWarning })
+    })
+  }
+}
+
+const logZISIBundleResult = ({ functionName, logs, result }) => {
+  const { bundlerErrors = [] } = result
+
+  if (bundlerErrors.length === 0) {
+    return
+  }
+
+  logErrorSubHeader(logs, `Fallback bundler used for function \`${functionName}\`:`)
+
+  bundlerErrors.forEach((bundlerError) => {
+    logBundleErrorObject({ logs, object: bundlerError })
```

Comment:
[sand] Would it make sense to create a `logBundleErrorObjects({ logs, objects })` method, to group the `forEach()` code into a single function?

If we pass a header string, this could also contain the `logErrorSubHeader()` call.

Reply:
Absolutely. I'll do that.

Reply:
Done in https://github.com/netlify/build/pull/2308/commits/156e122c163216b9f55f73e175cbf8edc967d8a5.